duckduckgo/Android
# Restricts what URLs originating from a subframe in the WebView can do

Task/Issue URL: https://app.asana.com/0/inbox/608920331025313/1148690082200045/1148578967949304
Tech Design URL: 
CC: 

**Description**:
Adds restrictions on what an iFrame can do when it's loading a URL

With this change, we only support 3rd party app launching if it was triggered from the main frame. For Android API 21-23, the functionality hasn't changed as we don't have information about whether on the main frame or not.

**Steps to test this PR**:
1. Test out the links in the linked Asana task on API 24+; verify we don't launch 3rd party apps
1. Test that links clicked on from the main frame still work as expected


---
###### Internal references:
[Software Engineering Expectations](https://app.asana.com/0/59792373528535/199064865822552)
[Technical Design Template](https://app.asana.com/0/59792373528535/184709971311943)
